sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2841,6 +2841,31 @@ object functions {
   // DateTime functions
   //////////////////////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * Creates a datetime interval
+   *
+   * @param years Number of years
+   * @param months Number of months
+   * @param weeks Number of weeks
+   * @param days Number of days
+   * @param hours Number of hours
+   * @param mins Number of mins
+   * @param secs Number of secs
+   * @return A datetime interval
+   * @group datetime_funcs
+   * @since 3.2.0
+   */
+  def make_interval(
```

Comment:
Can we expose in SparkR and PySpark too? If you're not used to it, we can run it separately too. cc @zero323 FYI

Reply:
I recommend creating a JIRA issue for the other langs and keep focusing on one language only. That will make code review easier (for one-language-only eyes like mine).

Reply:
@zero323 - are you OK if I create a separate JIRA and do a separate pull request to add this function to PySpark & R?  Seems like @HyukjinKwon is open to doing everything in one PR or having separate PRs.  @jaceklaskowski prefers one PR for language.  Just let me know how you'd like me to proceed to keep this moving forward, thanks!

Reply:
Yeah, we can run it separately.